pyscf/tools/trexio.py

```diff
@@ -223,3 +224,69 @@ def _group_by(a, keys):
     assert all(keys[:-1] <= keys[1:])
     idx = np.unique(keys, return_index=True)[1]
     return np.split(a, idx[1:])
+
+def get_occsa_and_occsb(mcscf, norb, nelec):
+    ci_coeff = mcscf.ci
+    num_determinants = int(np.sum(np.abs(ci_coeff) > 1e-8))
+    occslst = fci.cistring.gen_occslst(range(norb), nelec // 2)
+    selected_occslst = occslst[:num_determinants]
+
+    occsa = []
+    occsb = []
+    ci_values = []
+
+    for i in range(min(len(selected_occslst), mcscf.ci.shape[0])):
+        for j in range(min(len(selected_occslst), mcscf.ci.shape[1])):
+            ci_coeff = mcscf.ci[i, j]
+            if np.abs(ci_coeff) > 1e-8:  # Check if CI coefficient is significant
```

Comment:
Can we make this threshold for CI coefficients configurable as an argument to this function? I would trust the user to have the full control of the determinant filtering (we can keep your 1e-8 as default).

Reply:
@q-posev I am not sure if setting a CI threshold is a good idea. For example, after a sCI run, my system produces the following (full) list of determinants with CI values using a threshold of 1e-8:
```
occsa    occsb       CI coeff
[0 1]       [0 1]    -0.99779049148461
[0 2]       [0 2]     0.03340264667473
[0 2]       [1 2]    -0.03321881541135
[1 2]       [0 2]    -0.03321881541135
[1 2]       [1 2]     0.03303662972152
```

Now, let’s say a user wants to keep only values higher than 0.1. In this case, the code would register only [0 1] [0 1] -0.9977904914846 and set the others to 0. However, this corresponds to truncating the wavefunctions, which is a different process, right? This would also imply that the one- and two-electron integrals would need to be renormalized to account for the new threshold.

I chose a threshold of 1e-8, as I believe it’s sufficient to filter out coefficients that are effectively zero in terms of bit precision.

Let me know your thoughts.

Reply:
Done.
I decided to place the CI threshold argument right after the user-defined `nelec` value for consistency. I thought it might be counterintuitive for the user to place this threshold after the `trexio_file` argument.

Let me know your thoughts.